test/hooks.on-listen.test.js

```diff
@@ -1056,36 +1053,57 @@ test('async onListen does not need to be awaited', t => {
 
 test('onListen hooks do not block /1', t => {
   t.plan(2)
+
+  let timer
+  let listenDone = false
+  let doneRef
   const fastify = Fastify()
   t.teardown(fastify.close.bind(fastify))
 
   fastify.addHook('onListen', function (done) {
-    setTimeout(done, 500)
+    if (listenDone) {
```

Comment:
I don't think it is testing what it should be tested.
In most case `onListen` hook will run after the listen callback.
So, the timer should never be fired.

Reply:
The solution should be ensure the run order.
`onListen` MUST fire after the listen callback, so it is truth by it's claim never block listening.

Reply:
Maybe also calling an endpoint to contextualize the app is receiving requests disregarding the state of the `onListen` hook